### PR TITLE
Import scripts for course facilitators and workshop organziers

### PR DESCRIPTION
**Pull request by aoby**
Import course facilitators from a csv download of this spreadsheet: https://docs.google.com/spreadsheets/d/1ggv0WQp0xn3sYLs2n5g1RjgR3sC2VoyW-RrUTJf1edM/edit#gid=1713094616

This script doesn't have to be in production immediately, but I need to run it on production to populate the data for the facilitator summit.

### CODE REVIEW

File: dashboard/scripts/import_course_facilitators.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+COL_LAST_NAME = 'Last'
+COL_USER_ID = 'Code Studio ID'
+COL_EMAIL = 'Email Address'
+COL_COURSE = 'Program'
+
+facilitators_csv = ARGV[0]
+unless facilitators_csv
+  puts 'Usage: import_course_facilitators facilitators.csv'
+  exit
+end
```
Comment (mehalshah) [inline dashboard/scripts/import_course_facilitators.rb:13]: Optional for something this simple but option parser is nice

File: dashboard/scripts/import_course_facilitators.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+error_count = 0
+CSV.foreach(facilitators_csv, headers: true) do |row|
+  user_id = row[COL_USER_ID]
+  first_name = row[COL_FIRST_NAME]
+  last_name = row[COL_LAST_NAME]
+  email = row[COL_EMAIL]
+  course = row[COL_COURSE]
+
+  next unless email
```
Comment (mehalshah) [inline dashboard/scripts/import_course_facilitators.rb]: If the CSV is malformed, is it worth failing early? Will discuss offline

File: dashboard/scripts/import_course_facilitators.rb [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  next unless first_name || last_name
+
+  raise "Unrecognized course #{course}" unless Pd::Workshop::COURSES.include? course
+
+  facilitator =
```
Comment (mehalshah) [inline dashboard/scripts/import_course_facilitators.rb:33]: Similar question here - if the facilitator is nonexistent, is it worth continuing?

**Comment by mehalshah** [comment]
Looks fine but will discuss one thing offline

**Comment by aoby** [comment]
After offline discussion, I split this into 2 passes: validate then import.

File: dashboard/scripts/import_course_facilitators.rb [hunk 4]
```diff
@@ -37,29 +38,22 @@ def [](header)
...
     end
 
-  unless facilitator
-    STDERR.puts "Facilitator #{user_id} (#{email}) not found."
-    error_count += 1
-    next
-  end
-
   # validate other fields
   unless facilitator.name == "#{first_name} #{last_name}" && facilitator.email.casecmp(email)
```
Comment (mehalshah) [inline dashboard/scripts/import_course_facilitators.rb]: Make the name case insensitive? In fact, maybe go lowercase, strip out anything thats not a-z and compare that

File: dashboard/scripts/import_course_facilitators.rb [hunk 5]
```diff
@@ -37,29 +38,22 @@ def [](header)
...
 
-if error_count == 0
-  puts 'Success.'
+# Second pass, import the data.
+course_facilitator_params_list.each do |course_facilitator_params|
+  Pd::CourseFacilitator.find_or_create_by course_facilitator_params
 end
 
-exit error_count
+puts 'Success.'
```
Comment (mehalshah) [inline dashboard/scripts/import_course_facilitators.rb]: Indicate how many things you created.

**Comment by mehalshah** [comment]
Couple comments, then ship

**Comment by mehalshah** [comment]
LGTM

**Comment by aoby** [comment]
I added 2 more scripts to set up workshop organizers (plp and district), also from that spreadsheet. Please take a look. There's some common code here between the scripts so maybe next week I'll pull that out into another file and consolidate a bit.

**Comment by mehalshah** [comment]
LGTM - consolidate when you can

**Pull request merged**